modules/flowable-cmmn-engine/src/test/java/org/flowable/cmmn/test/tenant/ChangeTenantIdCaseInstanceTest.java

```diff
@@ -0,0 +1,264 @@
+        String caseInstanceIdCCompleted = startCase(TEST_TENANT_C, "caseWithMilestone", "caseInstanceCCompleted", true);
+        
+        // Starting case instances that will be kept active
+        String caseInstanceIdAActive = startCase(TEST_TENANT_A, "caseWithMilestone", "caseInstanceAActive", false);
+        String caseInstanceIdBActive = startCase(TEST_TENANT_B, "caseWithMilestone", "caseInstanceBActive", false);
+        String caseInstanceIdCActive = startCase(TEST_TENANT_C, "caseWithMilestone", "caseInstanceCActive", false);
+        
+        Set<String> caseInstanceIdsTenantA = Sets.newSet(caseInstanceIdACompleted, caseInstanceIdAActive);
+        Set<String> caseInstanceIdsTenantB = Sets.newSet(caseInstanceIdBCompleted, caseInstanceIdBActive);
+        Set<String> caseInstanceIdsTenantC = Sets.newSet(caseInstanceIdCCompleted, caseInstanceIdCActive);
+
+        // Prior to changing the Tenant Id, all elements are associated to the original tenant
+        checkTenantIdForAllInstances(caseInstanceIdsTenantA, TEST_TENANT_A, "prior to changing to " + TEST_TENANT_B);
+        checkTenantIdForAllInstances(caseInstanceIdsTenantB, TEST_TENANT_B, "prior to changing to " + TEST_TENANT_B);
+        checkTenantIdForAllInstances(caseInstanceIdsTenantC, TEST_TENANT_C, "prior to changing to " + TEST_TENANT_B);
+
+        // First we simulate the change
+        ChangeTenantIdResult simulationResult = cmmnManagementService
+                .createChangeTenantIdBuilder(TEST_TENANT_A, TEST_TENANT_B).simulate();
+
+        // All the instances should stay in the original tenant after the simulation
+        checkTenantIdForAllInstances(caseInstanceIdsTenantA, TEST_TENANT_A, "after simulating the change to " + TEST_TENANT_B);
+        checkTenantIdForAllInstances(caseInstanceIdsTenantB, TEST_TENANT_B, "after simulating the change to " + TEST_TENANT_B);
+        checkTenantIdForAllInstances(caseInstanceIdsTenantC, TEST_TENANT_C, "after simulating the change to " + TEST_TENANT_B);
+
+        // We now proceed with the changeTenantId operation for all the instances
+        ChangeTenantIdResult result = cmmnManagementService
+                .createChangeTenantIdBuilder(TEST_TENANT_A, TEST_TENANT_B).complete();
+
+        // All the instances should now be assigned to the tenant B
+        checkTenantIdForAllInstances(caseInstanceIdsTenantA, TEST_TENANT_B, "after the change to " + TEST_TENANT_B);
+        checkTenantIdForAllInstances(caseInstanceIdsTenantB, TEST_TENANT_B, "after the change to " + TEST_TENANT_B);
+
+        // The instances for tenant C remain untouched
+        checkTenantIdForAllInstances(caseInstanceIdsTenantC, TEST_TENANT_C, "after the change to " + TEST_TENANT_B);
+
+        // The simulation result must match the actual result
+        assertThat(simulationResult).isEqualTo(result).as("The simulation result must match the actual result.");
```

Comment:
Let's also verify the actual contents of the results. The reason for that is because we want to verify that the API is correct and contains the expected data. This applies to all the tests in this file and the other similar tests for the other engines.

Reply:
Actually, the isEqualTo is checking the contents of the result map. The equalTo function is overriden.

Reply:
Yes I saw that. But the `simulationResult` is not tested as well. Both can very much be empty and the assertion will be true.

Therefore, I prefer testing the contents more.

In addition to that in `assertThat(...).isEqualTo(...).as(...)` the `as` part is obsolete. If you want it to have an impact it should be before the `isEqualTo` e.g. `assertThat(...).as(...).isEqualTo(...)`

Reply:
I understand, @filiphr . I will apply the check of the contents in addition to the isEqualTo. Also, I will review the order of the .as() call.